### ISSUE TITLE
WebVMT: Review media block concept against WebVTT

### ISSUE THREAD

**Issue by rjksmith** [w3c/sdw#1099]
@tidoust wrote:

The [WebVMT media definition block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-definition-block) is not specific to maps, does not exist in WebVTT, and yet is defined as mandatory in WebVMT. That seems a good point to discuss with timed text guys if not already done.

**Comment by rjksmith** [comment]
As you rightly say, the [WebVMT media definition block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) is equally applicable to WebVTT, and is important for WebVMT due to the file lifecycle which was [discussed](
https://github.com/w3c/sdw/blob/gh-pages/proposals/geotagging/webvmt/meetings/2018-10-24-webvmt-breakout.md) in the breakout session - slide 4 of the [presentation](
http://w3c.github.io/sdw/proposals/geotagging/webvmt/meetings/2018-10-24_MetadataCues.pdf).

It's presence is mandatory for WebVMT to enable rapid indexing, so only the VMT file need be parsed during a search. Though not as essential to WebVTT, this could be a useful optional feature.

**Comment by tidoust** [comment]
> It's presence is mandatory for WebVMT to enable rapid indexing, so only the VMT file need be parsed during a search. Though not as essential to WebVTT, this could be a useful optional feature.

Why wouldn't the rapid indexing argument be as essential to WebVTT?

Re-reading the current draft, I note that the WebVMT media definition block is mandatory, but that it may be empty given that a [WebVMT media settings list](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-settings-list) "consists of **zero or more** of the following components". That is probably not what you want.

One reason I believe it is a good point to discuss with timed text guys is that this seems to go against the way tracks are being associated in the media world, be them in-band tracks or out-of-band tracks. For in-band tracks, tracks are associated within a media container. For out-of-band tracks, the `<video>` element acts as the container with `<track>` children linking text tracks to media tracks. In both cases, I believe this is done so that tracks may be reused in other contexts. For example, back to WebVMT, what about the following situations:
1. Someone records a trip to a foreign country to create a TV/Radio documentary (à la [J'irai dormir chez vous](https://en.wikipedia.org/wiki/J%27irai_dormir_chez_vous)). For TV, the documentary consists of a media file with a video and audio track, and of a WebVMT file. For radio, the documentary consists of a media file that only contains the audio track, and of the same WebVMT file. If the WebVMT file needs to link to a media file, which media file should it link to?
2. Same idea for situations where there are multiple cameras, creating multiple media files for the same media timeline. Which media file should a WebVMT file be associated with?

In other words, there are related media tracks (video, audio) that can produce different media files. For indexing and archival purpose, I'd want a way to associate a WebVMT file to these different media files, which is what the `<track>` element gives me in HTML.

Also, having the link at the WebVMT level seems error prone: it's easy to forget to update the link (e.g. if the name of the media file changes) and this would not have any noticeable effects for rendering (or would it?).

**Comment by rjksmith** [comment]
Media Timed Events Task Force (MTE TF) is focussed on rendering activities, rather than search, to the extent that the [Geolocation Rapid Indexing](https://github.com/w3c/me-media-timed-events/issues/6#issuecomment-422342705) use case I submitted was not included in the [MTE Editor's Draft](https://w3c.github.io/me-media-timed-events/#synchronized-rendering-of-web-resources). As the main purpose of the WebVMT media block is to enable rapid indexing, I see no compelling requirement in WebVTT currently.

The [WebVMT media block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-definition-block) is mandatory but may contain [no settings](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-settings-list) to ensure that it always remains visible, even in the few cases when it's not used, for exactly the reasons you mention.

My understanding is that WebVMT is an out-of-band solution in media terminology, as the VMT file is separate to the media file, so using the HTML `<track>` element to associate it with a `<video>` element is a consistent approach. For HTML purposes, the [WebVMT media block](
http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) is a 'hint', as I mentioned in the TPAC [breakout session](https://github.com/w3c/sdw/blob/gh-pages/proposals/geotagging/webvmt/meetings/2018-10-24-webvmt-breakout.md), so I'd expect the HTML association to take precedence.

I can think of two web use cases where the [WebVMT media information](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) is not provided by HTML:

   1. **Video archive:** A PHP script generates HTML pages on-the-fly from a collection of files in a video archive, which requires information to associate the correct VMT and media files.

   1. **Multiple file input:** Files loaded using `<input type="file" multiple>` are delivered in an arbitrary order. If more than two files are loaded, media and VMT files must be correctly paired.

I can understand the need for shared subtitle files where media files have common audio tracks, as the subtitle text is related to the audio. However, I don't see a comparable requirement for WebVMT as the VMT file relates to (and is typically created by) the video capture device, i.e. the camera, which is unique. Hence, I don't see a WebVMT use case for track reuse.

   1. **TV vs radio:** Creating separate media files for TV or radio is an editing activity, so it's not unreasonable to expect corresponding separate VMT files to be created at the same time by the editing suite for each media file produced.

   1. **Multiple camera angles:** Recording separate camera angles requires multiple cameras, each creating their own unique VMT file. Even though the moving object path may be common, the VMT files captured by each camera would be different by definition. For example:
        - Colocated cameras facing in different directions, e.g. forward- and rear-facing cameras on a vehicle.
        - Non-colocated cameras observing the same target, e.g. dashcam and roadside cameras.

If there is a need for a single VMT file to associate with multiple media files, a simple way to achieve this would be to allow multiple [WebVMT media block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) definitions within the same file and I'd be interested to know what the use case is.

The editing suite would be responsible for ensuring that the [WebVMT media block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) is correctly defined in the VMT file, though I agree that this could be a pitfall for manual editing.

**Comment by rjksmith** [comment]
**SDW IG Review Summary (5/12/18)**

The purpose of the [WebVMT media block](
http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media) is to identify the linked media file for out-of-band, i.e. non-embedded, data. This information is _known_ at creation by the capture device, _duplicated_ in HTML by the `<track>` element to display in a web page and _required_ to index a video archive.

Acknowledged weaknesses are:
   1. The information needs to be correctly maintained, e.g. when a media file is renamed.
   1. There could be a conflict in HTML between the details in the [WebVMT media url setting](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-url-setting) and in the `src` attribute of the `<source>` element.

_Conclusion_
   * Add an explanatory note to clarify the purpose of this information.
   * Add a note highlighting the need to maintain the link correctly.
   * Add a note explaining how a user agent should resolve conflicting information.
   * Raise a new issue to investigate a better solution for handling out-of-band links.

**Comment by rjksmith** [comment]
Updated Editor's Draft - pull request #1105:
   * Added a note to [WebVMT media definition block](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-definition-block) clarifying its purpose.
   * Added a note to [WebVMT media settings list](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-settings-list) explaining search engine use.
   * Updated [WebVMT media settings list](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-settings-list) note to highlight a pitfall when renaming files.
   * Updated [WebVMT media url setting](http://w3c.github.io/sdw/proposals/geotagging/webvmt/#dfn-webvmt-media-url-setting) note to resolve non-matching URLs.

**Comment by rjksmith** [comment]
Raised issue #1107.
